src/clj/rems/db/resource.clj

```diff
@@ -1,25 +1,58 @@
 (ns rems.db.resource
   (:require [rems.db.core :as db]
-            [rems.db.licenses :as licenses]))
-
-(defn- format-resource
-  [{:keys [id owneruserid modifieruserid organization resid enabled archived]}]
-  {:id id
-   :owneruserid owneruserid
-   :modifieruserid modifieruserid
-   :organization organization
-   :resid resid
-   :enabled enabled
-   :archived archived})
+            [rems.json :as json]
+            [rems.schema-base :as schema-base]
+            [schema.coerce :as coerce]
+            [schema.core :as s]))
+
+(s/defschema ResourceDb
+  {:id s/Int
+   :owneruserid s/Str
+   :modifieruserid s/Str
+   :organization schema-base/OrganizationId
+   :resid s/Str
+   :enabled s/Bool
+   :archived s/Bool
+   (s/optional-key :resource/duo) {(s/optional-key :duo/codes) [schema-base/DuoCode]}})
+
+(def ^:private coerce-ResourceDb
+  (coerce/coercer! ResourceDb coerce/string-coercion-matcher))
+
+(def ^:private validate-ResourceDb
+  (s/validator ResourceDb))
+
+(defn- format-resource [resource]
+  (let [resourcedata (json/parse-string (:resourcedata resource))]
+    (-> resource
+        (update :organization (fn [organization-id] {:organization/id organization-id}))
+        (dissoc :resourcedata)
+        (merge resourcedata))))
 
 (defn get-resource [id]
   (when-let [resource (db/get-resource {:id id})]
-    (format-resource resource)))
+    (-> resource
+        format-resource
+        coerce-ResourceDb)))
 
 (defn get-resources [filters]
   (->> (db/get-resources)
        (db/apply-filters filters)
-       (map format-resource)))
+       (map format-resource)
+       (map coerce-ResourceDb)))
```

Comment:
maybe pipe functions in one `map` instead of two sequential `map`s?

Reply:
maybe even as one function that is used for both `get-resource` and `get-resources`